spec/controllers/pieces_controller_spec.rb

```diff
@@ -147,13 +147,30 @@
 
       sign_in @player2
 
-      put :update, params: {id: black_queen.id, x_position: 1, y_position: 6, format: :js }
+      put :update, params: { id: black_queen.id, x_position: 1, y_position: 6, format: :js }
       black_queen.reload
       @game.reload
       expect(black_queen.x_position).to eq 1
       expect(black_queen.y_position).to eq 6
       expect(@game.winner).to eq @player2
     end
+
+    it 'puts game in check when black piece takes a white piece and puts white king in check' do
+      white_king = create(:king, x_position: 0, y_position: 4, game_id: @game.id, piece_number: 4)
+      white_pawn = create(:pawn, x_position: 0, y_position: 5, game_id: @game.id, piece_number: 5)
+      black_rook = create(:rook, x_position: 0, y_position: 7, game_id: @game.id, piece_number: 6)
+
+      sign_in @player2
+
+      black_rook.move_to!(0, 5)
+      white_pawn.reload
+      black_rook.reload
+      @game.reload
+
+      expect(black_rook.x_position).to eq 0
+      expect(black_rook.y_position).to eq 5
+      expect(@game.check?(!black_rook.is_white?)).to eq true
+    end
```

Comment:
Since this does not actually check the update function it should probably be moved to its own describe block. Also since the changes were to the `check_test` method should probably change line 172 to something along the lines of
`expect(check_test(black_rook, 0, 5).to eq true`
Perhaps an extra line of testing to also check for the correct flash message. This is going to be wrong but along the lines of:
`expect(flash.message).to eq 'White King in Check.'

To be frank looks like we missed the boat completely on testing the whole `check_test`. That is on all of us. This will tie into some earlier comment because as I review the method it seems like there is definitely some situations that could be a problem.

Reply:
I put this spec in update because it is testing functionality that is within the `piece#update` method. Unfortunately you cannot call `check_test` within this spec as it is a private method for `pieces_controller`. 

As far as the flash goes, I know you can check the message in a feature spec but testing for a flash message in controller spec is not so easy. I don't think it's necessary in this case either. We could write a feature spec to make sure it displays but the only thing I really care about here is that the game is in check (which means the flash does have the message, because our update action revolves around the flash messages).  

A lot of the `check_test` method is tested within `pieces_controller_spec.rb`. We don't have any specs explicitly for that method, but that method is heavily integrated in the `piece#update` action. Once again, we could write some tests but I don't think it is necessary. If you think of any specs that could cover a feature of the app we missed then I am all for it.

Reply:
By using `black_rook.move_to!(0, 5)` on line 165 you are completely avoiding the `check_test` method be called. It is only called if you run a put request. https://mixandgo.com/learn/3-ways-of-testing-private-methods-in-rails talks about some ways to test private methods in RSpec. All come with some pros and cons. And for the sake of simplicity we could just instead of using the `move_to!` method on line 165 we could use:
`put :update, params: { id: black_rook.id, x_position: 0, y_position: 5, format: :js }`
I do think this will actually lead to an error because it is not blacks turn so add this on line 162:
`move = create(:move, game_id: @game.id, piece_id: white_pawn.id, start_piece: 5)`
Not for sure if it works with this being entangled with JS but then can add:
`expect(flash[:alert]).to eq 'White King in Check.'`

Reply:
`put :update, params: { id: black_rook.id, x_position: 0, y_position: 5, format: :js }`
I tried this off the bat, but it doesn't seem to want to update the x and y position. 

I am not trying to test the `check_test` method with this spec though. I am simply trying to make sure the state of the game is in check after taking an enemy piece and putting the enemy king in check, which it is. 

`expect(flash[:alert]).to eq 'White King in Check.'` returns nil no matter if I use `move_to!` or the `put :update, params:` line.

Reply:
Yeah it is not updating the piece because it is whites turn.

Reply:
Okay makes sense with the JS. As you have stated elsewhere that is an easy check within a game and not really a game breaking bug either way. ie players would laugh if it said white king in check when it is really the black king but would not break the game from functioning.

Reply:
Ahh that makes sense. I will update the test and move the white piece, then the black piece.

Reply:
Nevermind, it seems that even if I move a white piece first, it is still not updating the black pieces moves.

Reply:
This spec is doing what I wanted it to, so I am going to leave it at that.